test/parallel_exec.jl

```diff
@@ -1463,3 +1463,18 @@ syms = setup_syms(3, workers())
 clear!(syms, workers())
 test_clear(syms, workers())
 
+# Test partial recovery from a deserialization error in CapturedException
+try
+    expr =
+    quote
+        type DontExistOn1
+            x
+        end
+        throw(BoundsError(DontExistOn1(1), 1))
+    end
+    remotecall_fetch(()->eval(expr), id_other)
```

Comment:
add `error("unexpected")` after this so it doesn't silently pass if the remotecall somehow doesn't throw